app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js

```diff
@@ -95,15 +92,27 @@ export default class NMRiumDisplayer extends React.Component {
       const eventDataType = eventData.type;
 
       if (eventDataType === 'nmr-wrapper:data-change') {
-        
         const nmrWrapperActionType = eventData.data.actionType;
         if (nmrWrapperActionType !== '') {
-          const nmriumData = eventData.data.data;
+          let nmriumData = null;
+          if (eventData.data) {
+            const { state } = eventData.data;
+            if (state) {
+              nmriumData = state;
+            }
+            else {
+              nmriumData = eventData.data;
+            }
+          }
```

Comment:
Hm, the whole `if` tree looks a complex to me, making it hard to follow:
![image](https://github.com/ComPlat/chemotion_ELN/assets/209056/2d492ab4-8b2c-4051-973a-a9995d70534e)

can we simplify the code to:

```
nmriumData = eventData.data?.state || eventData.data;
```

![image](https://github.com/ComPlat/chemotion_ELN/assets/209056/9dab0549-d6b5-471b-864f-c1204598f973)

This leads to the same results except that in the first two cases, we end up with undefined instead of null, which could probably further been worked around with `nmriumData = ( ....the line above... ) || null` in case it makes a difference later in the code.

Reply:
Yes, we can keep it simple. But I doubt is it clear to follow up, especially when we will update our site based on nmrium-wrapper returned data.